### PR TITLE
Remove ErrorPageOptions properties except SourceCodeLineCount

### PR DESCRIPTION
**Pull request by henkmollema**
Start of refactoring out the `ErrorPageOptions`

- Removed all properties except `SourceCodeLineCount`
- Updated ErroPageMiddleware (including some minor code-style changes)
- Updated CompilationErrorPage.cshtml and ErrorPage.cshtml
- Added `run` command to PageGenerator project.json to execute from command line

#142

### CODE REVIEW

**Comment by henkmollema** [comment]
cc @Tratcher

**Comment by henkmollema** [comment]
Also related: #80

**Comment by henkmollema** [comment]
cc @davidfowl @Eilon

File: src/Microsoft.AspNet.Diagnostics/ErrorPageMiddleware.cs [hunk 1]
```diff
@@ -128,31 +125,23 @@ private Task DisplayRuntimeException(HttpContext context, Exception ex)
         {
             var request = context.Request;
 
-            ErrorPageModel model = new ErrorPageModel()
-            {
-                Options = _options,
-            };
+            var model = new ErrorPageModel
+                        {
```
Comment (Tratcher) [inline src/Microsoft.AspNet.Diagnostics/ErrorPageMiddleware.cs]: nit spacing

File: src/Microsoft.AspNet.Diagnostics/ErrorPageMiddleware.cs [hunk 2]
```diff
@@ -128,31 +125,23 @@ private Task DisplayRuntimeException(HttpContext context, Exception ex)
...
-            if (_options.ShowQuery)
-            {
-                model.Query = request.Query;
-            }/* TODO:
-            if (_options.ShowCookies)
-            {
+            model.ErrorDetails = GetErrorDetails(ex, showSource: true).Reverse();
+            model.Query = request.Query;
+
+            /* TODO:
```
Comment (Tratcher) [inline src/Microsoft.AspNet.Diagnostics/ErrorPageMiddleware.cs]: This cookies code can be uncommented now. It might need some minor updates.

Comment (henkmollema) [inline src/Microsoft.AspNet.Diagnostics/ErrorPageMiddleware.cs]: What about the `Environment`? Not sure what it should do.

Comment (Tratcher) [inline src/Microsoft.AspNet.Diagnostics/ErrorPageMiddleware.cs]: We don't know either https://github.com/aspnet/Diagnostics/issues/135

Comment (henkmollema) [inline src/Microsoft.AspNet.Diagnostics/ErrorPageMiddleware.cs]: I see, I'll leave it in a comment.

**Comment by henkmollema** [comment]
@Tratcher updated

File: src/Microsoft.AspNet.Diagnostics/ErrorPageMiddleware.cs [hunk 3]
```diff
@@ -128,31 +125,18 @@ private Task DisplayRuntimeException(HttpContext context, Exception ex)
         {
             var request = context.Request;
 
-            ErrorPageModel model = new ErrorPageModel()
+            var model = new ErrorPageModel
             {
                 Options = _options,
+                ErrorDetails = GetErrorDetails(ex, showSource: true).Reverse(),
```
Comment (Tratcher) [inline src/Microsoft.AspNet.Diagnostics/ErrorPageMiddleware.cs]: do you still need this showSource parameter?

Comment (henkmollema) [inline src/Microsoft.AspNet.Diagnostics/ErrorPageMiddleware.cs]: Nah, I don't think it adds any value.

**Comment by henkmollema** [comment]
@Tratcher updated

**Comment by Tratcher** [comment]
:shipit:
I'll get this merged.

**Comment by henkmollema** [comment]
Cool, I'll squash and rebase in a sec.

**Comment by henkmollema** [comment]
Squashed/rebased

**Comment by Eilon** [comment]
Can some string resources such as ErrorPageHtml_EnableShowExceptions be removed as well?

https://github.com/aspnet/Diagnostics/blob/3fa5dee2d7f377d57e75fbb0d328cd26b6ff093a/src/Microsoft.AspNet.Diagnostics/Resources.resx#L136-L138

Not sure if there are others.

**Comment by henkmollema** [comment]
@Eilon good point, I'll take a look at it.

**Comment by henkmollema** [comment]
@Eilon looks like `ErrorPageHtml_EnableShowExceptions` was the only unused one.

/cc @Tratcher

**Comment by Eilon** [comment]
Looks good to me now, thanks!

**Comment by Tratcher** [comment]
Rebased and Merged

**Pull request closed without merging**